aries_cloudagent/wallet/keys/routes.py

```diff
@@ -0,0 +1,208 @@
+        },
+    )
+
+    verification_method = fields.Str(
+        data_key="verificationMethod",
+        required=False,
+        metadata={
+            "description": "Optional kid to bind to the keypair",
+            "example": "did:web:example.com#key-01",
+        },
+    )
+
+
+class CreateKeyResponseSchema(OpenAPISchema):
+    """Response schema from creating a new key."""
+
+    multikey = fields.Str(
+        metadata={
+            "description": "The Public Key Multibase format (multikey)",
+            "example": "z6MkgKA7yrw5kYSiDuQFcye4bMaJpcfHFry3Bx45pdWh3s8i",
+        },
+    )
+
+    kid = fields.Str(
+        metadata={
+            "description": "The associated kid",
+            "example": "did:web:example.com#key-01",
+        },
+    )
+
+
+class UpdateKeyRequestSchema(OpenAPISchema):
+    """Request schema for updating an existing key pair."""
+
+    multikey = fields.Str(
+        required=True,
+        metadata={
+            "description": "Multikey of the key pair to update",
+            "example": "z6MkgKA7yrw5kYSiDuQFcye4bMaJpcfHFry3Bx45pdWh3s8i",
+        },
+    )
+
+    verification_method = fields.Str(
+        data_key="verificationMethod",
+        required=True,
+        metadata={
+            "description": "New kid to bind to the key pair",
+            "example": "did:web:example.com#key-02",
+        },
+    )
+
+
+class UpdateKeyResponseSchema(OpenAPISchema):
+    """Response schema from updating an existing key pair."""
+
+    multikey = fields.Str(
+        metadata={
+            "description": "The Public Key Multibase format (multikey)",
+            "example": "z6MkgKA7yrw5kYSiDuQFcye4bMaJpcfHFry3Bx45pdWh3s8i",
+        },
+    )
+
+    kid = fields.Str(
+        metadata={
+            "description": "The associated kid",
+            "example": "did:web:example.com#key-01",
+        },
+    )
+
+
+class FetchKeyResponseSchema(OpenAPISchema):
+    """Response schema from updating an existing key pair."""
+
+    multikey = fields.Str(
+        metadata={
+            "description": "The Public Key Multibase format (multikey)",
+            "example": "z6MkgKA7yrw5kYSiDuQFcye4bMaJpcfHFry3Bx45pdWh3s8i",
+        },
+    )
+
+    kid = fields.Str(
+        metadata={
+            "description": "The associated kid",
+            "example": "did:web:example.com#key-01",
+        },
+    )
+
+
+@docs(tags=["wallet"], summary="Fetch key info.")
+@response_schema(FetchKeyResponseSchema, 200, description="")
+@tenant_authentication
+async def fetch_key(request: web.BaseRequest):
+    """Request handler for fetching a key.
+
+    Args:
+        request: aiohttp request object
+
+    """
+    context: AdminRequestContext = request["context"]
+    multikey = request.match_info["multikey"]
+    try:
+        key_info = await MultikeyManager(context.profile).from_multikey(
+            multikey=multikey,
+        )
+        return web.json_response(key_info, status=200)
```

Comment:
The status codes are done like this in many places, but you can also do `HTTPStatus.OK`. I think it's a bit better than just straight integers. You don't have to change it in this PR though.